spec/controllers/adopters_controller_spec.rb

```diff
@@ -318,13 +308,215 @@
+          expect(parse_response(response)['adopters']).to_not be_nil
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'de un adoptante' do
+      context 'con un usuario con permisos para editar adoptantes' do
+        before(:each) do
+          request.headers['X-USER-TOKEN'] = user.authentication_token
+          get :show, id: adopter.id, format: 'json'
+        end
+
+        it 'devuelve una array no vacio' do
+          expect(parse_response(response)).to_not be_nil
+        end
+
+        it 'devuelve una array con los datos del adoptante creado' do
+          expect(parse_response(response)['id']).to eq(adopter.id)
+          expect(parse_response(response)['ci']).to eq(adopter.ci)
+          expect(parse_response(response)['first_name']).to eq(adopter.first_name)
+          expect(parse_response(response)['last_name']).to eq(adopter.last_name)
+          expect(parse_response(response)['email']).to eq(adopter.email)
+          expect(parse_response(response)['phone']).to eq(adopter.phone)
+          expect(parse_response(response)['house_description']).to eq(adopter.house_description)
+          expect(parse_response(response)['blacklisted']).to eq(adopter.blacklisted)
+          expect(parse_response(response)['home_address']).to eq(adopter.home_address)
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+      end
+
+      context 'con un usuario sin permisos para editar adoptantes' do
+        before(:each) do
+          request.headers['X-USER-TOKEN'] = user2.authentication_token
+          get :show, id: adopter.id, format: 'json'
+        end
+
+        it 'devuelve una array no vacio' do
+          expect(parse_response(response)).to_not be_nil
+        end
+
+        it 'devuelve una array con los datos del adoptante creado' do
+          expect(parse_response(response)['id']).to eq(adopter.id)
+          expect(parse_response(response)['ci']).to eq(adopter.ci)
+          expect(parse_response(response)['first_name']).to eq(adopter.first_name)
+          expect(parse_response(response)['last_name']).to eq(adopter.last_name)
+          expect(parse_response(response)['email']).to eq(adopter.email)
+          expect(parse_response(response)['phone']).to eq(adopter.phone)
+          expect(parse_response(response)['house_description']).to eq(adopter.house_description)
+          expect(parse_response(response)['blacklisted']).to eq(adopter.blacklisted)
+          expect(parse_response(response)['home_address']).to eq(adopter.home_address)
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+      end
+    end
+  end
+
+  describe 'GET #search' do
+    context 'se listan los adoptantes buscados' do
+
+      let!(:adToSearch) { create(:adopter, first_name: 'Miguel', last_name: 'Viera') }
```

Comment:
Es necesario el `let!` aca? Si se puede sacar mejor, porque asi se crean las variables solamente cuando se van a usar, y no ante cada it